x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/hooks.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { IntegrationCardItem } from '@kbn/fleet-plugin/public';
+
+import { useNavigation } from '../../../../../common/lib/kibana';
+import { getFilteredCards } from './utils';
+import { INTEGRATION_TABS } from './const';
+
+export const useIntegrationCardList = ({
```

Comment:
Can we put this hook in its own file `use_integration_card_list.ts` along with all the related logic? Including the `getFilteredCards` function that currently is in the utils file, but is only used by this hook.